resources/documentation/monitoring/prometheus.rst

```diff
@@ -0,0 +1,594 @@
+* karma
+* ipmi_exporter
+* snmp_exporter
+* grafana
+
+Installation
+------------
+
+Prometheus Server
+^^^^^^^^^^^^^^^^^
+
+First create file
+/etc/bluebanquise/inventory/group_vars/all/addons/prometheus.yml with the
+following content:
+
+.. code-block:: yaml
+
+  prometheus:
+  scrape_interval: 1m
+  evaluation_interval: 2m
+  alertmanager:
+    group_wait: 1m
+    group_interval: 10m
+    repeat_interval: 3h
+
+.. seealso:: https://www.robustperception.io/whats-the-difference-between-group_interval-group_wait-and-repeat_interval
+
+Now, simply add to the playbook of your choice (which is for the Prometheus
+server) the prometheus_server role (change the values of enable_services and start_services accordingly):
+
+.. code-block:: yaml
+  
+  vars:
+    - enable_services: true
+    - start_services: true
+  roles:
+    - role: prometheus_server
+      tags: prometheus_server
+
+
+Then run:
+
+.. code-block:: text
+
+  ansible-playbook /etc/ansible/playbooks/<your server playbook> --tags prometheus_server
+
+Now prometheus_server should be installed and configured with a minimal
+configuration.
+
+The configuration file for Prometheus is located under
+/etc/prometheus/prometheus.yml.
+It contains all the exporters to scrape, and more.
+
+It is now time to configure client side. Note that while the
+/etc/bluebanquise/inventory/group_vars/all/addons/prometheus.yml file is only
+used by the prometheus_server role, the client files seen in next section will
+be shared by both server and client role. Server will use it to know which
+exporters to scrap on who, and clients will use it to know which exporters to
+install locally.
+
+Prometheus Client
+^^^^^^^^^^^^^^^^^
+
+For each *equipment_profile*, create a file called monitoring.yml that contains
+the desired exporters to be deployed, into equipment_profile folder of your
+target hosts groups.
+
+For example, to set exporters to be scrapped and installed on all hosts of
+equipment_profile equipment_XX, create the file
+/etc/ansible/inventory/group_vars/equipment_XX/monitoring.yml, with the
+following content:
+
+.. code-block:: yaml
+
+  monitoring:
+    exporters:
+      node_exporter:
+        package: node_exporter
+        service: node_exporter
+        port: 9100
+      ha_cluster_exporter:
+        package: ha_cluster_exporter
+        service: ha_cluster_exporter
+        scrape_interval: 10m
+        scrape_timeout: 4m
+        port: 9664
+
+This will setup here two exporters on these equipments: node_exporter and
+ha_cluster_exporter.
+
+Also don't forget to add the name of the package you want to install and the
+service name.
+
+.. note::
+  As you can see, you can also add the scrape_interval (which is how
+  often the metrics get scraped), and the scrape_timeout (which represents how
+  long until a scrape request times out).
+
+.. note::
+  If you want to add exporters, make sure your package contains the
+  binary and the .service file, put preferably under /usr/local/bin and
```

Comment:
why preferably in /usr/local/bin?

Reply:
all packaged exporters provided by BB have their binaries located under /usr/local/bin, so we know where they all are

Reply:
rpm -ql <package> can provide the location of the files.
I understand however.